extensions/amazon-lambda-resteasy/runtime/src/main/java/io/quarkus/amazon/lambda/resteasy/runtime/container/ResteasyLambdaContainerHandler.java

```diff
@@ -71,15 +88,84 @@ protected AwsHttpServletResponse getContainerResponse(AwsProxyHttpServletRequest
+        try {
+            nettyDispatch(connection, request, response);
+        } catch (Exception e) {
+            response.setStatus(HttpStatus.SC_INTERNAL_SERVER_ERROR);
+        } finally {
+            connection.close();
+        }
+    }
+
+    private void nettyDispatch(VirtualClientConnection connection, AwsProxyHttpServletRequest request,
+            AwsHttpServletResponse response) throws Exception {
+        String path = request.getPathInfo();
 
-        httpServletRequest.setServletContext(getServletContext());
+        String query = request.getQueryString();
+        if (query != null) {
+            path = path + '?' + query;
+        }
+
+        String host = request.getRemoteHost();
+        if (request.getRemotePort() != -1) {
+            host = host + ':' + request.getRemotePort();
+        }
 
-        doFilter(httpServletRequest, httpServletResponse, null);
-        Timer.stop("RESTEASY_HANDLE_REQUEST");
+        DefaultHttpRequest nettyRequest = new DefaultHttpRequest(HttpVersion.HTTP_1_1,
+                HttpMethod.valueOf(request.getMethod()), path);
+        nettyRequest.headers().set("Host", host);
+        for (Map.Entry<String, List<String>> header : request.getAwsProxyRequest().getMultiValueHeaders().entrySet()) {
+            nettyRequest.headers().add(header.getKey(), header.getValue());
+        }
+
+        HttpContent requestContent = LastHttpContent.EMPTY_LAST_CONTENT;
+        if (request.getAwsProxyRequest().getBody() != null) {
+            String body = request.getAwsProxyRequest().getBody();
+            ByteBuf buffer = Unpooled.wrappedBuffer(body.getBytes());
+            requestContent = new DefaultLastHttpContent(buffer);
+        }
+
+        connection.sendMessage(nettyRequest);
+        connection.sendMessage(requestContent);
+        ByteArrayOutputStream baos = null;
+        boolean last = false;
+        while (!last) {
+            Object msg = connection.queue().poll(100, TimeUnit.MILLISECONDS);
+            if (msg != null) {
+                try {
+                    if (msg instanceof HttpResponse) {
+                        HttpResponse res = (HttpResponse) msg;
+
+                        response.setStatus(res.status().code());
+                        for (Map.Entry<String, String> entry : res.headers()) {
+                            response.addHeader(entry.getKey(), entry.getValue());
+                        }
+                    }
+                    if (msg instanceof HttpContent) {
+                        HttpContent content = (HttpContent) msg;
+                        if (baos == null) {
+                            baos = new ByteArrayOutputStream(500);
+                        }
+                        ByteBuf buffer = content.content();
+
+                        while (buffer.readableBytes() > 0) {
+                            byte[] bytes = new byte[buffer.readableBytes()];
+                            buffer.readBytes(bytes);
```

Comment:
io.netty.buffer.ByteBuf#readBytes(java.io.OutputStream, int) should be more efficient here.